--- /usr/local/lib/python3.5/site-packages/PyInstaller/depend/bindepend.py	2016-12-01 20:48:08.000000000 +0900
+++ bindepend.py	2016-12-01 20:45:30.000000000 +0900
@@ -622,13 +622,20 @@
             #    '../lib\x00\x00')
             cmd_type = command[0].cmd
             if cmd_type == LC_RPATH:
-                rpath = str(command[2])
+                #rpath = str(command[2])
+                rpath = command[2].decode('utf-8')
                 # Remove trailing '\x00' characters.
                 # e.g. '../lib\x00\x00'
                 rpath = rpath.rstrip('\x00')
+                # Replace the @executable_path and @loader_path keywords
+                # with the actual path to the binary.
+                executable_path = os.path.dirname(pth)
+                rpath = rpath.replace('@executable_path', executable_path)
+                rpath = rpath.replace('@loader_path', executable_path)
                 # Make rpath absolute. According to Apple doc LC_RPATH
                 # is always relative to the binary location.
-                rpath = os.path.normpath(os.path.join(os.path.dirname(pth), rpath))
+                #rpath = os.path.normpath(os.path.join(os.path.dirname(pth), rpath))
+                rpath = os.path.normpath(os.path.join(executable_path, rpath))
                 run_paths.update([rpath])
             else:
                 # Frameworks that have this structure Name.framework/Versions/N/Name
